### ISSUE TITLE
IgnoreTypes doesn't ignore enumerable types

### ISSUE THREAD

**Issue by breyed** [EntityGraphQL/EntityGraphQL#181]
If an entity contains a property of type `List<T>`,  `AddAllFields` tries to add it even if `SchemaBuilderOptions.IgnoreTypes` contains `T`. This causes the error:
> No schema type found for dotnet type T. Make sure you add it or add a type mapping

``` c#
class E {
    public List<T> Ts { get; set; }
}

eSchemaType.AddAllFields(new SchemaBuilderOptions { IgnoreTypes = new[] { nameof(T) }.ToHashSet() });
```

EntityGraphQL version: 3.0.0

**Comment by lukemurray** [comment]
Will be fixed in 3.0.3.

Annoyingly `IgnoreTypes` wasn't used before 3.0.0 but it was missed in my review as it really should be a list of `Type`s! But that is a breaking change. 

FYI, 3.1 will mark this as obsolete and introduce a `List<Type>` to use.

**Comment by lukemurray** [comment]
Fixed in 3.0.3

**Comment by breyed** [comment]
Thanks for the tip on needing to use `FullName`. I'm looking forward to change to use Type instead of string. IMHO, it's a simple enough change that it's not worth the complexity of having two lists of ignored types, one with strings, and the other with Types. You can just change the type to `List<Type>`. The compiler will readily flag the calling code that needs to be changed. If that means a 4.x happens to quickly follow 3.x, it's not a problem at all.
